Source/WebKit/UIProcess/Cocoa/WKShareSheet.mm

```diff
@@ -251,12 +254,17 @@ - (void)presentWithShareDataArray:(NSArray *)sharingItems inRect:(std::optional<
             [self dismiss];
     }];
 
-    UIPopoverPresentationController *popoverController = [_shareSheetViewController popoverPresentationController];
-    if (rect) {
-        popoverController.sourceView = webView;
-        popoverController.sourceRect = *rect;
-    } else
-        popoverController._centersPopoverIfSourceViewNotSet = YES;
+    if (_customPresentationStyle != UIModalPresentationNone) {
+        _shareSheetViewController.get().allowsCustomPresentationStyle = YES;
+        _shareSheetViewController.get().modalPresentationStyle = _customPresentationStyle;
+    } else {
+        UIPopoverPresentationController *popoverController = [_shareSheetViewController popoverPresentationController];
+        if (rect) {
+            popoverController.sourceView = webView;
+            popoverController.sourceRect = *rect;
+        } else
+            popoverController._centersPopoverIfSourceViewNotSet = YES;
```

Comment:
Maybe it would be nice to also call this logic if `_customPresentationStyle == UIModalPresentationPopover`?